src/inputs/labeled-inputs.scss

```diff
@@ -28,169 +45,165 @@
       cursor: not-allowed;
     }
   }
+  // #endregion
 
-  > .iui-label {
-    margin-bottom: $iui-component-offset;
-    font-weight: $iui-font-weight-semibold;
-
-    &.iui-required {
-      &::after {
-        content: ' *';
-        @include themed {
-          color: t(iui-color-foreground-negative);
-        }
-      }
-    }
+  // Appropriate spacing even if no text label is given
+  .iui-checkbox,
+  .iui-radio {
+    min-height: $iui-line-height;
   }
 
-  > .iui-message {
-    display: flex;
-    width: fit-content;
-    align-items: center;
-    font-size: $iui-font-size-small;
-    line-height: $iui-line-height;
-    margin-top: $iui-component-offset;
-    @include themed {
-      color: t(iui-text-color-muted);
-    }
+  &.iui-inline-label {
+    grid-template: 'label inputs icon' / auto 1fr auto;
+    grid-row-gap: 0;
 
-    svg {
-      width: $iui-icons-default;
-      height: $iui-icons-default;
-      display: flex; // needed for vertical alignment of webcomponents
-      margin-right: $iui-xs;
-      @include themed {
-        fill: t(iui-icons-color);
-      }
+    &.iui-with-message {
+      grid-template:
+        'label inputs inputs'
+        '. icon message' / auto auto 1fr;
     }
 
-    > .iui-borderless {
-      border-top-left-radius: 0;
-      border-bottom-left-radius: 0;
+    > .iui-input-group {
+      display: flex;
+      gap: $iui-m;
 
-      svg {
-        @include themed {
-          fill: t(iui-icons-color-actionable);
+      > .iui-checkbox,
+      > .iui-radio,
+      > .iui-toggle-switch {
+        margin-right: $iui-m;
```

Comment:
Shouldn't this be for non-last children?

Reply:
I think you are correct.  I commented out the `gap` CSS to see what it would look like if only relying on `margin-right`. Notice the extra space between the last checkbox & status icon.

As is:
![Screen Shot 2021-09-14 at 4 00 56 PM](https://user-images.githubusercontent.com/849817/133325612-b45997d1-5aae-4778-8ea7-ce98585466f7.png)

After removing `gap`:
![Screen Shot 2021-09-14 at 4 01 27 PM](https://user-images.githubusercontent.com/849817/133325669-283f4bb1-6531-49ca-a7c5-174c4f400abd.png)

Reply:
I see there is some issues with IE so let's leave it as it is.